backend/engine/plugins/cicd_tools/detectors/pattern_detector.py

```diff
@@ -53,6 +53,8 @@ def _get_matches(self, path: str) -> list[Path]:
         base = Path(path)
 
         for pattern in self.patterns:
-            result.extend(base.rglob(pattern))
-
+            try:
+                result.extend(base.rglob(pattern))
+            except OSError:
+                continue
```

Comment:
Why don't we append to the `errors` list here?

Reply:
I meant `PatternDetector` in general. We could, for example, modify this function to return a tuple including a list of strings to add to errors/alerts.